### ISSUE TITLE
Anime 'S2' Season notation not being picked up from disk and other naming issues

### ISSUE THREAD

**Issue by ZadesSC** [Sonarr/Sonarr#1951]
## Seasons naming in episodes with S#
Animes using the S# for season numbers (ex. [HorribleSubs] Saekano S2 - 1 [720p].mkv) can get picked up from indexers and downloaded fine but if I already have the files on the disk in a the correct folder directory it doesn't get picked up.  It's fine if Sonarr puts them in the directory from a download client but if you delete the files, remove the entry and re populate the files and have Sonarr scan the directory it won't pick them up.

## Multi Part when there is no subgroup to prefix 
Also sometimes in the search it becomes listed as a multipart, which I'm guessing the S2 - 06 becmes read like episodes 2 to 6?  Found that with Saekano S2 - 06 (1280x720 HEVC2 EAC3) entry during the search.
![image](https://cloud.githubusercontent.com/assets/8403355/26569224/2d79b88c-44be-11e7-94ae-e50e93f1d996.png)

EDIT:  I've been playing around with it and it seems that its using different logic for grabbing from torrents/nzb compared to scanning disk.

## Issues with picking up episodes if season 2 names differ from series name, sonarr also ignores the seasons folder structure
Anime shows with a differing named season 2 can be grabbed from indexers but cannot be scanned from disk.  It also seems to not respect the Season folders on disk when this occurs.

Using Nyaruko: Crawling With Love! as an example(http://thetvdb.com/?tab=series&id=257521):
I have the following files for this anime:
 
![image](https://cloud.githubusercontent.com/assets/8403355/26573940/69ed081c-44d4-11e7-9ef5-e84c724de9fe.png)

I added the anime through Add Series on Sonarr and pointed it to this directory.  It the proceeded to only add the Season 2 files, and only to Season 1 episodes, completely ignoring the Season folder structure.

This is the end result of adding the anime and having it scan the disk
![image](https://cloud.githubusercontent.com/assets/8403355/26574052/c952e7c2-44d4-11e7-8866-d5c87f621079.png)

These are the episodes that it thinks are part of season 1.  Notice that it picks them up from the season 2 subfolder, completely ignoring the structure.
![image](https://cloud.githubusercontent.com/assets/8403355/26574109/fc836fc2-44d4-11e7-8b3a-bf184f211177.png)

This is most likely due to the fact that the name for the 2nd season of this anime is different from the first and so the sub groups decided to use the season 2 name.  Sonarr also detected a higher quality episodes in the season 2 folder over the first season and decided to ignore the folder structure.

## Some thoughts
I can see that allowing different season names might be non-trivial, but I think Sonarr should at least respect the seasons sub folder structure.  There is also the inconstant issue that Sonarr will detect and is able to pull the episodes with different season names from indexers (both nzb and torrents) )but are unable to import them from the harddrive.

**Comment by Taloth** [comment]
Sonarr has special regular expressions to parse anime titles. And this releases isn't named as such (usually there a [FanSub] prefix or suffix)
I'll leave to @markus101 to decide whether we want to do something about this edgecase.

**Comment by ZadesSC** [comment]
What about the issue with S2 naming convention not being picked up from disk?

**Comment by ZadesSC** [comment]
Added more info.  I think a easy fix would be to have Sonarr respect the season sub-folders if the episodes inside does not have a season value in its file name. So then Sonarr will parse episodes within the Season 2 sub-folder as season 2 episodes.

**Comment by markus101** [comment]
Sonarr expects files in the series folder to follow TVDB numbering, which in the case you pointed out above is not the case, it's using anidb numbering inside a TVDB folder structure (duplicate absolute episode numbers differentiated by the series title).

If we end up using the folder to pull in more information I think that would be covered by: https://github.com/Sonarr/Sonarr/issues/457 which would help solve https://github.com/Sonarr/Sonarr/issues/903 as well.

For `Saekano S2 - 06 (1280x720 HEVC2 EAC3)`, that's a poorly named release, it's not getting parsed as a range of episodes, but a full season pack because nothing in the release name triggers the anime parsing, this might be something that https://github.com/Sonarr/Sonarr/issues/457 could also help with, but it might not (it's still a hard to parse release name).

**Comment by ZadesSC** [comment]
I also think that adding a manual file/folder import option as per #1941 would allow users to fix this by themselves by forcing Sonarr to recognize those files as the correct episodes.  I want to have Sonarr at least recognize the files as episodes and the only way to do so currently would be to manually search and download each individual episode and point it to the torrent (if the torrent if already completed, Sonarr will just add the completed download to the folder).  I'm assuming this works because trackers and newsgroups and jackett offer additional metadata on the files that are not in the files themselves.

**Comment by markus101** [comment]
I think https://github.com/Sonarr/Sonarr/issues/305 would make more sense than #1941 - I see #1941 as a way to open manual import from the series details page rather than having to go to the Wanted: Missing page and passing the series information in could help automate linking it to the correct series, but in this case it wouldn't add anything as the absolute episode number conflicting causes problems.

**Comment by markus101** [comment]
Closing as this issue is covered by a couple existing issues and made some notes on #457
